xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -162,16 +153,17 @@ public String toString() {
    * Updates the resource subscription for the given resource type.
    */
   // Must be synchronized.
-  void adjustResourceSubscription(ResourceType type) {
+  void adjustResourceSubscription(XdsResourceType<? extends ResourceUpdate> resourceType) {
```

Comment:
Just use `XdsResourceType<?>`; the "extends ResourceUpdate" is extraneous. When defining it with a name like T (like below with `<T extends ResourceUpdate> void ackResponse(`) you do need the extends because the T is defined outside the scope of XdsResourceType itself. But with the ? here, you don't.